pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -245,6 +255,60 @@ func (r *ReconcileOneAgent) buildDynatraceClient(instance *dynatracev1alpha1.One
 	return dtc, err
 }
 
+func (r *ReconcileOneAgent) reconcileNodesMarkedForDeletion(
+	reqLogger logr.Logger,
+	instance *dynatracev1alpha1.OneAgent,
+	dtc dtclient.Client) error {
+
+	// nodes := &corev1.NodeList(metav1.ListOptions{})
+	nodeList := &corev1.NodeList{}
+	fieldSelector := fields.SelectorFromSet(fields.Set{
+		"spec.unschedulable": "True",
+	})
+	labelSelector := labels.SelectorFromSet(labels.Set(instance.Spec.NodeSelector))
+	listOps := &client.ListOptions{
+		//--field-selector=spec.unschedulable=True
+		FieldSelector: fieldSelector,
+		LabelSelector: labelSelector,
+	}
+	err := r.client.List(context.TODO(), listOps, nodeList)
+	if err != nil {
+		reqLogger.Error(err, "failed to list nodes", "listops", listOps)
+		return err
+	}
+
+	for _, node := range nodeList.Items {
+
+		cordoned := node.Spec.Unschedulable
+
+		nodeInternalIP := getInternalIPForNode(node)
+		reported, ok := cordonedNodes[nodeInternalIP]
+
+		// cordoned and not in map
+		if cordoned && !ok {
+			cordonedNodes[nodeInternalIP] = false
+
+			// cordoned, in map, but not reported
+			if !reported {
+				status, err := dtc.PostMarkedForTerminationEvent(nodeInternalIP)
+				if err != nil {
+					reqLogger.Error(err, "reconcileNodesMarkedForDeletion: error sending event")
+					return err
+				}
+				reqLogger.Info("reconcileNodesMarkedForDeletion: event sent, status %s", status)
+
+				// cordoned, in map, and reported
+				cordonedNodes[nodeInternalIP] = true
+			}
+		}
+		// if uncordoned, but present in map => remove from map
+		if !cordoned && ok {
```

Comment:
Hmm, since we are filtering by `fieldSelector: spec.unschedulable=true`, are we going to get any uncordoned nodes?

Reply:
Should we remove the field selector and iterate over all the nodes instead?

Reply:
Yes, I think that is the way to go. We already go through every node to look for the agent versions anyway.